src/lib/DateInput.svelte

```diff
@@ -127,11 +127,61 @@
       visible = false
     }
   }
+
+  /** Lets the date popup positions its self, to best fit on the screen*/
+  export let dynamicPositioning = false
+
+  let InputElement: HTMLInputElement
+  let pickerElement: HTMLElement | null
+  let showAbove = false
+  let pickerLeftPosition: number | null = null
+
+  function setDatePickerPosition() {
+    // Defaults
+    showAbove = false
+    pickerLeftPosition = null
+
+    if (visible && pickerElement && dynamicPositioning) {
+      // The child of the dateField is what is visually seen, all calculations should use this to make sure they line up properly
+      const inputRect = InputElement.getBoundingClientRect()
+      const widowWidth = window.innerWidth
+      const rightPickerOverflow = pickerElement.offsetWidth - inputRect.width
+
+      const bottomThreshold = inputRect.bottom + pickerElement.offsetHeight + 5
+      const rightThreshold = inputRect.left + pickerElement.offsetWidth + 5
+
+      if (bottomThreshold > window.innerHeight) {
+        // If .date-time-field is on the bottom half of the screen, open above
+        showAbove = true
+      }
+      if (rightThreshold > widowWidth) {
+        // If date-time-field is on the right of the screen, open to the left
+        pickerLeftPosition = -rightPickerOverflow
+      }
```

Comment:
I made it so it's only positioned above or left to prevent overflow. So if the `<input`> is in the bottom half, the picker still shows up below if there's space.

Do you think that's ok?

Reply:
That would work, I think this is more of a personal preference. I just prefer when it always opens towards the center of the screen, but if you prefer they way you have it I am fine with that as its your project.